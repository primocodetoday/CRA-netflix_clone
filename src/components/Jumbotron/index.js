import React from 'react';
import PropTypes from 'prop-types';
import {
  Item,
  Inner,
  Container,
  Pane,
  Title,
  SubTitle,
  Image,
} from './styles/jumbotron';

const Jumbotron = ({ children, direction }, ...restProps) => {
  return (
    <Item {...restProps} direction={direction}>
      <Inner>{children}</Inner>
    </Item>
  );
};

Jumbotron.Container = function JumbotronContainer({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Jumbotron.Pane = function JumbotronPane({ children, ...restProps }) {
  return <Pane {...restProps}>{children}</Pane>;
};

Jumbotron.Title = function JumbotronTitle({ children, ...restProps }) {
  return <Title {...restProps}>{children}</Title>;
};

Jumbotron.SubTitle = function JumboSubtronTitle({ children, ...restProps }) {
  return <SubTitle {...restProps}>{children}</SubTitle>;
};

Jumbotron.Image = function JumbotronImage({ ...restProps }) {
  return <Image {...restProps} />;
};

Jumbotron.propTypes = {
  direction: PropTypes.string,
};

Jumbotron.defaultProps = {
  direction: 'row',
};

export default Jumbotron;
