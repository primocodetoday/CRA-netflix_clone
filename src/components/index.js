export { default as Footer } from 'components/Footer';
export { default as Accordion } from 'components/Accordion';
export { default as Jumbotron } from 'components/Jumbotron';
