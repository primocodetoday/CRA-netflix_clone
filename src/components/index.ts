export { Accordion } from 'components/Accordion/Accordion';
export { Banner } from 'components/Banner/Banner';
export { Card } from 'components/Card/Card';
export { CTAForm } from 'components/CTAForm/CTAForm';
export { Form } from 'components/Form/Form';
export { Footer } from 'components/Footer/Footer';
export { Header } from 'components/Header/Header';
export { Jumbotron } from 'components/Jumbotron/Jumbotron';
export { Loading } from 'components/Loading/Loading';
export { Player } from 'components/Player/Player';
export { Profiles } from 'components/Profiles/Profiles';
