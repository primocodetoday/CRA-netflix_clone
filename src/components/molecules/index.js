export { default as Footer } from 'components/molecules/Footer/Footer';
export { default as Accordion } from 'components/molecules/Accordion/Accordion';
export { default as Jumbotron } from 'components/molecules/Jumbotron/Jumbotron';
export { default as CTAForm } from 'components/molecules/CTAForm/CTAForm';
export { default as Header } from 'components/molecules/Header/Header';
