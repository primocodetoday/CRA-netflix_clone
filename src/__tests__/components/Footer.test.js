import React from 'react';
import { render } from '@testing-library/react';
import { Footer } from 'components';

describe('<Footer /> renders properly', () => {
  it('with data', () => {
    const { container, getByText } = render(
      <Footer>
        <Footer.Title>Questions?</Footer.Title>
        <Footer.Break />
        <Footer.Row>
          <Footer.Column>
            <Footer.Link href="#">FAQ</Footer.Link>
          </Footer.Column>
        </Footer.Row>
        <Footer.Text>Netflix Poland</Footer.Text>
      </Footer>,
    );

    expect(getByText('Questions?')).toBeTruthy();
    expect(getByText('FAQ')).toBeTruthy();
    expect(getByText('Netflix Poland')).toBeTruthy();

    expect(container.firstChild).toMatchSnapshot();
  });
});
