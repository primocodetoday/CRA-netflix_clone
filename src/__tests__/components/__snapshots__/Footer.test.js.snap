// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> renders properly with data 1`] = `
<div
  class="StyledFooter__Container-sc-1hptcbq-0 eiiqpZ"
>
  <div
    class="StyledFooter__Title-sc-1hptcbq-5 uHgyO"
  >
    Questions?
  </div>
  <div
    class="StyledFooter__Break-sc-1hptcbq-3 lldsVx"
  />
  <div
    class="StyledFooter__Row-sc-1hptcbq-2 eJA-dWP"
  >
    <div
      class="StyledFooter__Column-sc-1hptcbq-1 cGoXWq"
    >
      <a
        class="StyledFooter__Link-sc-1hptcbq-4 eeXysR"
        href="#"
      >
        FAQ
      </a>
    </div>
  </div>
  <p
    class="StyledFooter__Text-sc-1hptcbq-6 cWeGfp"
  >
    Netflix Poland
  </p>
</div>
`;
